samples/hybrid-connections/java/simple-http-demo/src/main/java/HttpListener.java

```diff
@@ -0,0 +1,67 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Scanner;
+
+import com.microsoft.azure.relay.HybridConnectionListener;
+import com.microsoft.azure.relay.RelayConnectionStringBuilder;
+import com.microsoft.azure.relay.RelayedHttpListenerResponse;
+import com.microsoft.azure.relay.TokenProvider;
+
+public class HttpListener {
+	static final String CONNECTION_STRING_ENV_VARIABLE_NAME = "RELAY_CONNECTION_STRING";
+	static final RelayConnectionStringBuilder connectionParams = new RelayConnectionStringBuilder(
+			System.getenv(CONNECTION_STRING_ENV_VARIABLE_NAME));
+
+	public static void main(String[] args) throws URISyntaxException {
+		TokenProvider tokenProvider = TokenProvider.createSharedAccessSignatureTokenProvider(
+				connectionParams.getSharedAccessKeyName(),
+				connectionParams.getSharedAccessKey());
+		HybridConnectionListener listener = new HybridConnectionListener(
+				new URI(connectionParams.getEndpoint().toString() + connectionParams.getEntityPath()),
+				tokenProvider);
+
+		// The "context" object encapsulates both the incoming request and the outgoing response
+		listener.setRequestHandler((context) -> {
+			String receivedText = "";
+			if (context.getRequest().getInputStream() != null) {
+				try (Reader reader = new BufferedReader(new InputStreamReader(context.getRequest().getInputStream(), "UTF8"))) {
+					StringBuilder builder = new StringBuilder();
+					int c = 0;
+					while ((c = reader.read()) != -1) {
+						builder.append((char) c);
+					}
```

Comment:
This seems burdensome. Is there a simpler way to read a Stream into a String?

Reply:
Ugh, there may not be. Have I mentioned I like .NET :)

---
In reply to: [267902758](https://github.com/Azure/azure-relay/pull/41#discussion_r267902758) [](ancestors = 267902758)

Reply:
This is kind of standard for reading off a input stream :( We can change it, but I am personally not too concerned with it

---
In reply to: [267903734](https://github.com/Azure/azure-relay/pull/41#discussion_r267903734) [](ancestors = 267903734,267902758)